cli/cli/commands/kurtosis_context/context_switch/switch.go

```diff
@@ -57,6 +56,21 @@ func run(ctx context.Context, _ *flags.ParsedFlags, args *args.ParsedArgs) error
 		return stacktrace.NewError("An error occurred retrieving current context prior to switching to the new one '%s'", contextIdentifier)
 	}
 
+	if !store.IsRemote(contextPriorToSwitch) {
+		engineManager, err := engine_manager.NewEngineManager(ctx)
+		if err != nil {
+			return stacktrace.Propagate(err, "An error occurred creating an engine manager.")
+		}
+		engineStatus, _, _, err := engineManager.GetEngineStatus(ctx)
+		if err != nil {
+			return stacktrace.Propagate(err, "An error occurred retrieving the engine status.")
+		}
+		if engineStatus == engine_manager.EngineStatus_Running {
+			logrus.Infof("An engine is running locally. Stop the engine before switching context so the remote engine port can be forwarded locally.")
```

Comment:
I would rephrase this as:
Stop the engine prior to switching context by running `kurtosis engine stop`